File: sunflower-core/src/main/java/org/tdf/sunflower/TransactionPoolConfig.java
package org.tdf.sunflower;

import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@ConfigurationProperties(prefix = "sunflower.transaction-pool")
@Component
@Setter
public class TransactionPoolConfig {
    private long expiredIn;

    public long getExpiredIn() {
        return expiredIn <= 0 ? 300 : expiredIn;
    }
}

File: sunflower-core/src/main/java/org/tdf/sunflower/TransactionPoolConfig.kt
package org.tdf.sunflower

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@ConfigurationProperties(prefix = "sunflower.transaction-pool")
@Component
data class TransactionPoolConfig(
    var expiredIn: Long = 0,
    var lockTimeout: Long = 0,
)